src/main/java/org/example/repository/ProductRepository.java

```diff
@@ -2,24 +2,90 @@
 
 import org.example.entity.Product;
 
-public class ProductRepository implements CrudRepository<Product> {
+import java.sql.*;
+
+public abstract class ProductRepository implements CrudRepository<Product> {
```

Comment:
Don't make it abstract. the fact that is's glowing red should tell you that you did something wrong. In this case you added additional methods to the interface and didn't implement them. Remove those methods and remove abstract keyword.